Resync web-platform-tests/digital-credentials from upstream

marcoscaceres opened the pull request:
#### 7758af84802a72abd0c902c620952704c80d2137
<pre>
Resync web-platform-tests/digital-credentials from upstream
<a href="https://bugs.webkit.org/show_bug.cgi?id=276707">https://bugs.webkit.org/show_bug.cgi?id=276707</a>
<a href="https://rdar.apple.com/problem/131904467">rdar://problem/131904467</a>

Reviewed by Abrar Rahman Protyasha.

Resync web-platform-tests/digital-credentials from Web Platform Tests.

Fixed DigitalCredentialsCoordinatorProxy.cpp to prevent an ASSERT crash when responding without an error message.

Upstream commit: <a href="https://github.com/web-platform-tests/wpt/commit/b6341be345c433eee87d7e1ae04778a5e87c2f75">https://github.com/web-platform-tests/wpt/commit/b6341be345c433eee87d7e1ae04778a5e87c2f75</a>

* LayoutTests/imported/w3c/resources/import-expectations.json:
* LayoutTests/imported/w3c/resources/resource-files.json:
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/META.yml: Added.
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/allow-attribute.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/allow-attribute.https.html:
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/dc-types.ts: Added.
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/default-permissions-policy.https.sub.html:
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/digital-credentials-static-methods.tentative.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/digital-credentials-static-methods.tentative.https.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/disabled-by-permissions-policy.https.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/disabled-by-permissions-policy.https.sub.html:
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/get-user-activation.https.html:
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/identity-create.tentative.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/identity-create.tentative.https.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/identity-get.tentative.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/identity-get.tentative.https.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/non-fully-active.https.html:
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/override-permissions-policy.https.sub-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/override-permissions-policy.https.sub.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/override-permissions-policy.https.sub.html.headers: Removed.
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/support/helper.js: Added.
(export.makeGetOptions):
(makeIdentityRequestProvider):
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/support/iframe.html:
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/support/w3c-import.log: Added.
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/tsconfig.json: Added.
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/w3c-import.log: Added.
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/digital-credentials/disabled-by-permissions-policy.https.sub-expected.txt: Copied from LayoutTests/imported/w3c/web-platform-tests/digital-credentials/disabled-by-permissions-policy.https.sub-expected.txt.
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/digital-credentials/identity-get.tentative.https-expected.txt: Added.
* Source/WebKit/UIProcess/DigitalCredentials/DigitalCredentialsCoordinatorProxy.cpp:
(WebKit::DigitalCredentialsCoordinatorProxy::requestDigitalCredential):

Canonical link: <a href="https://commits.webkit.org/282612@main">https://commits.webkit.org/282612@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1f2103859f2fd39ba50e201daf79acb4b5eee04b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/63630 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/42986 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/16227 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/67651 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/14238 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/50674 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/14518 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/51277 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/9842 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/66699 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/39884 "Passed tests") | [❌ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/55092 "Found 1 new API test failure: TestWebKitAPI.VideoControlsManager.StartPlayingLargestVideoInViewport (failure)") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/31906 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/36566 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/12471 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/13111 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/58070 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/12794 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/69347 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/7577 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/12365 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/58549 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/7610 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/55185 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/58764 "Found 3 new API test failures: /WebKitGTK/TestWebKitWebContext:/webkit/WebKitWebContext/uri-scheme, /WebKitGTK/TestContextMenu:/webkit/WebKitWebView/popup-event-signal, /WebKitGTK/TestContextMenu:/webkit/WebKitWebView/context-menu-key (failure)") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/14088 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/6312 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/38807 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/39886 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/40998 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/39629 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas commented on LayoutTests/imported/w3c/web-platform-tests/digital-credentials/get-user-activation.https.html:
Are there no test expectations for this one?

aprotyas replied:
Oh, I suppose the expectation hasn't changed?

aprotyas commented on LayoutTests/imported/w3c/web-platform-tests/digital-credentials/identity-get.tentative.https.html:
```diff
@@ -0,0 +1,195 @@
+    }
+  }, "navigator.identity.get() API rejects if there are no providers.");
+
+  promise_test(async (t) => {
+    iframeSameOrigin.focus();
+    const { contentWindow: iframeWindow } = iframeSameOrigin;
+    for (const provider of [undefined, []]) {
+      const options = makeGetOptions(provider);
+      await test_driver.bless("user activation");
+      await promise_rejects_js(
+        t,
+        iframeWindow.TypeError,
+        iframeWindow.navigator.identity.get(options)
+      );
+    }
+  }, "navigator.identity.get() API rejects if there are no providers for same-origin iframe.");
```
It looks like we're consistently failing this test on iOS simulator! Given the exception thrown, I believe `iframeSameOrigin.focus()` isn't behaving as expected?

```
+FAIL navigator.identity.get() API rejects if there are no providers for same-origin iframe. promise_rejects_js: function "function() { throw e }" threw object "NotAllowedError: The document is not focused." ("NotAllowedError") expected instance of function "function TypeError() {
+    [native code]
+}" ("TypeError")
```

aprotyas replied:
Alternative to fixing is to file a follow up FIXME bugzilla and check-in the failing expectations for iOS, I suppose.

marcoscaceres replied:
>  I believe iframeSameOrigin.focus() isn't behaving as expected?

That's correct. There's some issues with focus()'ing iframes in webkit generally, I believe. I've seen people trying to do things like this:
https://gist.github.com/erkobridee/e567d22180b9118229441f71791f6ec7

I couldn't find bug for it, so might just file one.

aprotyas reviewed (commented):
r+ but let's discuss the implications of the `/digital-credentials/identity-get.tentative.https.html` failure on iOS first.

aprotyas reviewed (approved):
Looks like we're still failing `imported/w3c/web-platform-tests/digital-credentials/disabled-by-permissions-policy.https.sub.html` on iOS because of a small expectation mismatch.

Can you make this change before landing?

```diff
-FAIL Setting digital-credentials-get=(self) disallows the API in same-origin iframes. assert_false: Digital Credential API expected false got true
+PASS Setting digital-credentials-get=(self) disallows the API in same-origin iframes.
```

marcoscaceres committed:
Resync web-platform-tests/digital-credentials from upstream
https://bugs.webkit.org/show_bug.cgi?id=276707
rdar://problem/131904467

Reviewed by Abrar Rahman Protyasha.

Resync web-platform-tests/digital-credentials from Web Platform Tests.

Fixed DigitalCredentialsCoordinatorProxy.cpp to prevent an ASSERT crash when responding without an error message.

Upstream commit: https://github.com/web-platform-tests/wpt/commit/b6341be345c433eee87d7e1ae04778a5e87c2f75

* LayoutTests/imported/w3c/resources/import-expectations.json:
* LayoutTests/imported/w3c/resources/resource-files.json:
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/META.yml: Added.
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/allow-attribute.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/allow-attribute.https.html:
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/dc-types.ts: Added.
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/default-permissions-policy.https.sub.html:
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/digital-credentials-static-methods.tentative.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/digital-credentials-static-methods.tentative.https.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/disabled-by-permissions-policy.https.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/disabled-by-permissions-policy.https.sub.html:
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/get-user-activation.https.html:
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/identity-create.tentative.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/identity-create.tentative.https.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/identity-get.tentative.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/identity-get.tentative.https.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/non-fully-active.https.html:
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/override-permissions-policy.https.sub-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/override-permissions-policy.https.sub.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/override-permissions-policy.https.sub.html.headers: Removed.
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/support/helper.js: Added.
(export.makeGetOptions):
(makeIdentityRequestProvider):
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/support/iframe.html:
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/support/w3c-import.log: Added.
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/tsconfig.json: Added.
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/w3c-import.log: Added.
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/digital-credentials/disabled-by-permissions-policy.https.sub-expected.txt: Copied from LayoutTests/imported/w3c/web-platform-tests/digital-credentials/disabled-by-permissions-policy.https.sub-expected.txt.
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/digital-credentials/identity-get.tentative.https-expected.txt: Added.
* Source/WebKit/UIProcess/DigitalCredentials/DigitalCredentialsCoordinatorProxy.cpp:
(WebKit::DigitalCredentialsCoordinatorProxy::requestDigitalCredential):

Canonical link: https://commits.webkit.org/282612@main
